### DISCUSSION TITLE
Windows version

### DISCUSSION THREAD

**Discussion by Aspire-Design-Ltd** [VK2BEA/HP8753-Companion#5] [General]
Hi.
I have a working version of your code running on windows.
The changes are minimal and it should be able to combine it with the Linux version.

The version I have running works with the ni488 version 17 driver which means you can use the NI GPIB-USB-B usb interface without any configuration setup.

Regards, MPC

**Comment by VK2BEA** [comment]
Wow, that's outstanding! 
Its interesting on a couple of fronts...  I knew GTK+ was available for windows and would be interested in finding out what changes were needed. Also, it's great to know that it indeed can run with the NI driver. The Linux GPIB driver **appears** to be NI compatible but I've not tried it.  (I have an NI USB-HS (although it could be a Chinese clone))
If you can #ifdef the changes into the code, then I'd be happy to incorporate them into the repo.

I've got a change that I'm working on to add a 'project' grouping for the configurations and traces, so that the drop-down lists don't become too large.

cheers,
  Michael

**Comment by Aspire-Design-Ltd** [comment]
Hi Micheal,
I feel you've done a great job with the code. It would be a great shame for windows users to miss out.
Only required minor changes to the source code to make it work for windows and I can list them for inclusion.

The main change is the build system to use. For windows I used visual code, cmake and minGW as this is currently a popular solution, although I have little experience with them.
I also built the Linux version with visual code and cmake. The main issues I had was finding the libraries to be included with the linker. I made direct absolute references to the libraries in the cmake file so that I was in control of what was linked.

So how do you feel about changing the build system so that it can be built on Linux or Windows? Visual Code is a convenient IDE where you download the project file and press go. I am not sure how easy it is for installation and publishing though.

Greatly appreciate you making this open source. Regards MPC.

**Comment by Aspire-Design-Ltd** [comment]
Hi Michael, I have been using the software for some real antenna measurements and it has performed exceptionally.

You have done a really good job! The high resolution smith charts are so.... COOL and give a very professional look!

MPC.

**Comment by Ernie49** [comment]
I have downloaded all files into same dirctory. Running under Windows 11 x64 I get error messages for libcairo-2.dll, libgdk_pixbuf-2.0.0.dll, libgdk-3.0.dll and libgio-2.0.0.dll: libxxx.dll not suited for running under Windows or has an error (short form, translated from German). Would be very happy, if this issue can be corrected, so I can use this program.
Thanks for your hard work so far.
Regards from Germany
Ernst

**Reply by VK2BEA** [reply]
Hello Ernie, the program here is for a Linux system; however, apparently it has been ported to Windows. I suggest you try to contact the commenter (above) who did this. He may be able to share his modifications.

**Reply by Aspire-Design-Ltd** [reply]
See --> https://github.com/Aspire-Design-Ltd/HP8753-Companion-Windows

I have been using this windows port with WIN11 for over 12 months. Its a couple of versions behind.

MPC.
